flows/actions/send_msg.go

```diff
@@ -128,27 +174,65 @@ func (a *SendMsgAction) Execute(run flows.Run, step flows.Step, logModifier flow
 // for message actions that specidy a template, this generates the template message where the message content should be
 // considered just a preview of how the template will be evaluated by the channel
 func getTemplateMsg(action *SendMsgAction, run flows.Run, urn urns.URN, channelRef *assets.ChannelReference, templateTranslation *flows.TemplateTranslation, evaluatedAttachments []utils.Attachment, evaluatedQuickReplies []string, unsendableReason flows.UnsendableReason, logEvent flows.EventCallback) *flows.MsgOut {
-	localizedVariables, _ := run.GetTextArray(uuids.UUID(action.Templating.UUID), "variables", action.Templating.Variables, nil)
+	evaluatedParams := make(map[string][]string)
+
+	templateTranslationParams := templateTranslation.Params()
 
-	evaluatedVariables := make([]string, len(localizedVariables))
-	for i, variable := range localizedVariables {
-		sub, err := run.EvaluateTemplate(variable)
-		if err != nil {
-			logEvent(events.NewError(err))
+	if _, ok := templateTranslationParams["body"]; ok {
```

Comment:
I'm confused by this... how does this work when variables is set but params isn't?

Reply:
`templateTranslationParams` in the params for WA templates that gives us a hint we should have variables for the components

```
{
  "uuid": "b7b35db9-d914-4cd3-a170-46eab02531bb",
  "params": {
    "body": [{"type": "text"}, {"type": "text"}],
    "button.0": [{"type": "text"}]
  }
}
```

so we use that and then check the old variables since the editor do not yet have support for setting params in templating

Even when the editor does below we wil overwrite the body params